cylc/flow/job_file.py

```diff
@@ -322,6 +321,8 @@ def _write_script(cls, handle, job_conf):
     @staticmethod
     def _write_epilogue(handle, job_conf):
         """Write epilogue."""
-        handle.write('\n\n. "${CYLC_DIR}/job.sh"\ncylc__job__main')
+        handle.write('\n\nTDIR="$(mktemp -d)"')
+        handle.write('\ncylc get-pkg-resources "${TDIR}" etc/job.sh')
```

Comment:
Slightly concerned about the efficiency cost of doing this for each job. It might be worth considering installing `job.sh` into the `cylc-run` dir(s) at suite run.

Reply:
Yeah, I thought this might get flagged.  I doubt that it's a problem, but hard to know under large scale heavy load conditions.
```
(venv) [oliverh@niwa-1007885 cylc-flow]$ time cylc get-pkg-resources /tmp etc/job.sh
Extracting etc/job.sh to /tmp/etc/job.sh

real    0m0.147s
user    0m0.119s
sys     0m0.029s
```

Reply:
OK next commit installs `job.sh` to `<run-dir>/.service/etc`:
 - on suite host, during suite registration (when the service dir is created).
 - or job hosts, during "remote init" 

I think is a more elegant solution anyway.

Reply:
> I think this is a more elegant solution anyway.

Agreed, it also provides a more guaranteed consistency within a suite run.